listener/pkg/event/watcher_event.go

```diff
@@ -33,25 +37,32 @@ func UnmarshalSKREvent(req *http.Request) (*unstructured.Unstructured, *Unmarsha
 		return nil, &UnmarshalError{"contract version cannot be empty", http.StatusBadRequest}
 	}
 
-	body, err := io.ReadAll(req.Body)
+	limitedReader := &io.LimitedReader{R: req.Body, N: requestSizeLimit}
+	body, err := io.ReadAll(limitedReader)
 	if err != nil {
```

Comment:
if you face an EOF here, maybe you should add a request size limited err. What I am a bit wondering is in your web server (which ever you are using) it is probably better conceptually to implement this as a middleware. WDYT?

Reply:
How should I encounter here an EOF? Since I am already using `ReadAll` and if you have a look into its definition:
```
// ReadAll reads from r until an error or EOF and returns the data it read.
// A successful call returns err == nil, not err == EOF. Because ReadAll is
// defined to read from src until EOF, it does not treat an EOF from Read
// as an error to be reported.
```
This reads until EOF and then returns, thus you will not be able to encounter here an EOF error.

Thus, and due to the reason this is the only usecase (receiving and distributing) Watcher events, I do not think so that we need to implement this as a middleware.

Reply:
There are many caveats here. First you also need to change the response code when you encounter Size limit to 413 (Request Entity Too Large). Also you should not continue with your code if you face it but rather abort. Currently since you use io.ReadAll it will continue as you described and error during the json.Unmarshal call. (since the json is truncated). I encourage you to write a proper middleware, for more orientation, see https://github.com/labstack/echo/blob/master/middleware/body_limit.go

Reply:
The requested refactoring will be implemented in a follow-up issue: https://github.com/kyma-project/runtime-watcher/issues/92